model/model_training/custom_datasets/ranking_collator.py

```diff
@@ -0,0 +1,64 @@
+import random
+from dataclasses import dataclass
+from typing import Iterable, Optional, Union
+
+from transformers.tokenization_utils_base import PaddingStrategy, PreTrainedTokenizerBase, TruncationStrategy
+
+
+@dataclass
+class RankingDataCollator:
+    """
+    Data collator that will dynamically pad the inputs for multiple choice received.
+    """
+
+    tokenizer: PreTrainedTokenizerBase
+    padding: Union[bool, str, PaddingStrategy] = True
+    max_length: Optional[int] = None
+    random_offset_probability: Optional[float] = 0.5
+    pad_to_multiple_of: Optional[int] = None
+
+    def process_one(self, messages: Iterable[str], return_length: bool = False):
+        messages = list(messages)
+
+        if random.random() < self.random_offset_probability:
+            truncation = TruncationStrategy.DO_NOT_TRUNCATE
+            max_length = None
+        else:
+            truncation = TruncationStrategy.LONGEST_FIRST
+            max_length = self.max_length
```

Comment:
Should we always truncate the begging for RM training?

Reply:
good point, IMO for RM we need to make sure to always include the last message (e.g. the assistant reply).